From 69a281c29ae8bd1041c2e7f23cc4f7f640a74559 Mon Sep 17 00:00:00 2001
From: Haohui Mai <wheat9@apache.org>
Date: Wed, 18 Mar 2015 11:08:08 -0700
Subject: [PATCH 1062/2848] HDFS-7953. NN Web UI fails to navigate to paths
 that contain #. Contributed by kanaka kumar
 avvaru.

(cherry picked from commit b547e6d6e0eae36107dc00791b56fd8cbbe6a714)

Change-Id: Ie0a3a3238fecdbc17549931dc8a523211e914cdd
---
 .../hadoop-hdfs/src/main/webapps/hdfs/explorer.js  |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
index 0a53dcd..131b2aa 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
@@ -78,6 +78,12 @@
     return data.RemoteException !== undefined ? data.RemoteException.message : "";
   }
 
+  function encode_path(abs_path) {
+    abs_path = encodeURIComponent(abs_path);
+    var re = /%2F/g;
+    return abs_path.replace(re, '/');
+  }
+
   function view_file_details(path, abs_path) {
     function show_block_info(blocks) {
       var menus = $('#file-info-blockinfo-list');
@@ -102,12 +108,6 @@
       menus.change();
     }
 
-    function encode_path(abs_path) {
-      abs_path = encodeURIComponent(abs_path);
-      var re = /%2F/g;
-      return abs_path.replace(re, '/');
-    }
-
     abs_path = encode_path(abs_path);
     var url = '/webhdfs/v1' + abs_path + '?op=GET_BLOCK_LOCATIONS';
     $.get(url).done(function(data) {
@@ -149,7 +149,7 @@
         return chunk.write('' + new Date(Number(value)).toLocaleString());
       }
     };
-    var url = '/webhdfs/v1' + dir + '?op=LISTSTATUS';
+    var url = '/webhdfs/v1' + encode_path(dir) + '?op=LISTSTATUS';
     $.get(url, function(data) {
       var d = get_response(data, "FileStatuses");
       if (d === null) {
-- 
1.7.9.5

